src/components/generic/movies/movies.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react'
+import { useRef, useEffect, FC } from 'react'
+import * as Movies from './style'
+import MovieCard from './movieCard'
+import ReactPlaceholder from 'react-placeholder'
+import 'react-placeholder/lib/reactPlaceholder.css'
+import { ModalProvider } from '../../App'
+import useBottom from '../../hooks/useBottom'
+import BounceLoader from 'react-spinners/BounceLoader'
+import { Movie } from '../../home/slider/sliderModal'
+
+interface Props {
+  movies: Movie[];
+  loader: boolean;
+  loading: boolean;
+}
+
+const MoviesPanel: FC<Props> = ({ movies, loader, loading }) => {
+
+  const renderLoading = (loading: boolean) =>
+    Array.from(new Array(20), (v, ii) => (
+      <ReactPlaceholder
+        key={ii}
+        type="rect"
+        ready={!loading}
+        color="#E0E0E0"
+        showLoadingAnimation
+        style={placeholderStyle}
+      >
+        <></>
+      </ReactPlaceholder>
+    ))
+
+
+  const RenderMovies = () => {
```

Comment:
I suggest declaring this outside of the `MoviesPanel` component or declaring it as a component on its own and pass props into it so React can optimize it.

Reply:
Done